selectors/views/channel_sidebar.ts

```diff
@@ -76,6 +74,67 @@ export function makeGetCurrentlyDisplayedChannelsForTeam(): (state: GlobalState,
             }).flat();
         },
     );
+})();
+
+// getUnreadChannels returns an array of all unread channels on the current team for display with the unread filter
+// enabled. Channels are sorted by recency with channels containing a mention grouped first.
+export const getUnreadChannels = (() => {
```

Comment:
Some of this logic could have been reused from the old sidebar, but that has a lot of cruft in it, so I chose to rewrite what I could. In particular, the sorting logic is copied from the old sidebar, but this one has a lot fewer dependencies than that one does, and it doesn't unnecessarily rewrite channel objects for no reason.